### DISCUSSION TITLE
How to open existing file or use template if not exist?

### DISCUSSION THREAD

**Discussion by wealthychef1** [SilentVoid13/Templater#1126] [Help] [answered]
Thanks for checking this out.  I apologize if this is in a FAQ but I haven't seen the answer yet after searching. 
I also was torn about which forum to put this in.  General Obsidian community or Templater Issues?  

## Current Behavior and Problem
I have a template for piano practice.  I use an obsidian URI to create a new file, which then automatically gets populated by Templater, like this: 

```
obsidian://advanced-uri?vault=wealthyvault&filepath=Planning%2520and%2520Organization%252FPeriodic%2520Practices%252FPiano%2520Stufdy%2520Blocks%252FUntitled.md
```

In my template I have this line of Templater code to change the filename to one that matches the current date:  

```
<% await(tp.file.rename(tp.date.now("YYYY-MM-DD")+" Piano Study")) %>
```

This works great, the first time.  :-). If I dare to attempt this twice in a day, I get a collision, as the file Templater is trying to make already exists at that point for the second attempt. 

## What I Want
- I would like to be able to check if the file exists before creating a file from a template. 
- I can do this with a shell script, but is there a more "pure" way through Obsidian?  
- Cheers and thanks

**Comment by Zachatoo** [comment, marked as answer]
I've got a few examples [here](https://zachyoung.dev/snippets/templater#create-file-if-it-doesnt-exist), though not sure if they'd work with the advanced URI plugin. It's the advanced URI plugin creating the file, not Templater. So you could check to see if the file exists, if it does, then delete the file that the advanced URI plugin created and just open the existing file instead.

I haven't tested this myself, but here's the basic idea.

```js
<%*
const fileName = tp.date.now("YYYY-MM-DD") + " Piano Study";
const existing = tp.file.find_tfile(fileName);
if (existing) {
  // delete file created by advanced URI plugin if desired file name does exist
  await app.vault.delete(tp.file.find_tfile(tp.file.title));
  // open existing file
  await app.workspace.getLeaf().openFile(existing);
} else {
  // rename file to desired file name if desired file name doesn't exist
  await tp.file.rename(fileName);
}
_%>
```

**Reply by wealthychef1** [reply]
Thank you, that's a very useful snippet.  And those examples are golden too.
